src/resolve/resolved-mdns.c

```diff
@@ -292,59 +325,93 @@ int manager_mdns_ipv6_fd(Manager *m) {
         if (m->mdns_ipv6_fd >= 0)
                 return m->mdns_ipv6_fd;
 
-        m->mdns_ipv6_fd = socket(AF_INET6, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
-        if (m->mdns_ipv6_fd < 0)
+        r = socket(AF_INET6, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (r < 0) {
+                log_error_errno(r, "mDNS-IPv6: Failed to create socket: %m");
                 return -errno;
+        }
+        m->mdns_ipv6_fd = r;
 
         r = setsockopt(m->mdns_ipv6_fd, IPPROTO_IPV6, IPV6_UNICAST_HOPS, &ttl, sizeof(ttl));
         if (r < 0) {
+                log_error_errno(r, "mDNS-IPv6: Failed to set IPV6_UNICAST_HOPS: %m");
                 r = -errno;
                 goto fail;
         }
 
-        /* RFC 4795, section 2.5 recommends setting the TTL of UDP packets to 255. */
+        /* RFC 4795, section 2.5 recommends Failed to set the TTL of UDP packets to 255. */
```

Comment:
This new version does not appears grammatically or semantically incorrect. "Failed" does not set anything...